galley/pkg/config/analysis/analyzers/virtualservice/destinationhosts.go

```diff
@@ -134,10 +135,29 @@ func initServiceEntryHostMap(ctx analysis.Context) map[util.ScopedFqdn]*v1alpha3
 		}
 		return true
 	}
+	extractSvc := func(r *resource.Instance) bool {
+		s := r.Message.(*corev1.ServiceSpec)
+		var se *v1alpha3.ServiceEntry
+		hostsNamespaceScope := string(r.Metadata.FullName.Namespace)
+		var ports []*v1alpha3.Port
+		for _, p := range s.Ports {
+			ports = append(ports, &v1alpha3.Port{
+				Number:   uint32(p.Port),
+				Name:     p.Name,
+				Protocol: string(p.Protocol),
+			})
+		}
+		host := util.ConvertHostToFQDN(r.Metadata.FullName.Namespace, r.Metadata.FullName.Name.String())
+		se = &v1alpha3.ServiceEntry{
+			Hosts: []string{host},
+			Ports: ports,
+		}
+		result[util.NewScopedFqdn(hostsNamespaceScope, r.Metadata.FullName.Namespace, r.Metadata.FullName.Name.String())] = se
+		return true
+	}
 
-	ctx.ForEach(collections.IstioNetworkingV1Alpha3Serviceentries.Name(), extractFn)
-	ctx.ForEach(collections.IstioNetworkingV1Alpha3SyntheticServiceentries.Name(), extractFn)
-
+	ctx.ForEach(collections.IstioNetworkingV1Alpha3Serviceentries.Name(), extractSe)
```

Comment:
`extractSe` and `extractSvc` are only used once. The lambda was useful before since the same function applied to real and synthetic SE. The following is more idiomatic with other analyzers. 

```golang
ctx.ForEach(collections.IstioNetworkingV1Alpha3Serviceentries.Name(), func(r *resource.Instance) bool {
    // ...
}
```